pkg/machine/e2e/cp_test.go

```diff
@@ -1,18 +1,210 @@
 package e2e_test
 
 import (
+	"archive/tar"
+	"bytes"
 	"fmt"
+	"io/fs"
 	"os"
+	"path"
 	"path/filepath"
 	"runtime"
+	"time"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("podman machine cp", func() {
-	It("all tests", func() {
+var _ = Describe("run cp commands", func() {
+	It("podman cp", func() {
+		var (
+			file = "foo.txt"
+
+			directory = "foo-dir"
+
+			fileInDirectory = "bar.txt"
+
+			stdinFile = "file.txt"
+
+			stdinDirectory = "stdin-dir"
+		)
+
+		sourceDir, err := os.MkdirTemp("", "host-source")
+		Expect(err).ToNot(HaveOccurred())
+
+		destinationDir, err := os.MkdirTemp("", "host-destination")
+		Expect(err).ToNot(HaveOccurred())
+
+		f, err := os.Create(filepath.Join(sourceDir, file))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.Close()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Get the file stat to check permissions later
+		sourceFileStat, err := os.Stat(filepath.Join(sourceDir, file))
+		Expect(err).ToNot(HaveOccurred())
+
+		err = os.MkdirAll(filepath.Join(sourceDir, directory), 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Get the directory stat to check permissions later
+		sourceDirStat, err := os.Stat(filepath.Join(sourceDir, directory))
+		Expect(err).ToNot(HaveOccurred())
+
+		f, err = os.Create(filepath.Join(sourceDir, directory, fileInDirectory))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.Close()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Get the file in directory stat to check permissions later
+		sourceFileInDirStat, err := os.Stat(filepath.Join(sourceDir, directory, fileInDirectory))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("copy from host to container")
+		name := randomString()
+		i := new(initMachine)
+		session, err := mb.setName(name).setCmd(i.withImage(mb.imagePath).withNow()).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		bm := basicMachine{}
+		newImgs, err := mb.setCmd(bm.withPodmanCommand([]string{"pull", TESTIMAGE})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(newImgs).To(Exit(0))
+		Expect(newImgs.outputToStringSlice()).To(HaveLen(1))
+
+		createAlp, err := mb.setCmd(bm.withPodmanCommand([]string{"create", TESTIMAGE, "top"})).run()
```

Comment:
one thing that would be nice would be to name this container `C` and then try to copy by name as well to ensure the IsAbs() logic is actually working? Or does that still not work.

Reply:
I ended up adding a container named C as part of a new test to validate container-to-container copying, so there's not an explicit test for the `IsAbs()` logic, but it does get validated as a side-effect.